Source/Core/getStringFromTypedArray.js

```diff
@@ -42,20 +44,128 @@ define([
+        var i, cp;
+        var codePoints = [];
+        for (i = 0; i < length; ++i) {
+            cp = utf8Handler(view[i]);
+            if (defined(cp)) {
+                codePoints.push(cp);
+            }
+        }
+
+        length = codePoints.length;
+        for (i = 0; i < length; ++i) {
+            cp = codePoints[i];
+            if (cp <= 0xFFFF) {
+                result += String.fromCharCode(cp);
+            } else {
+                cp -= 0x10000;
+                result += String.fromCharCode((cp >> 10) + 0xD800,
+                    (cp & 0x3FF) + 0xDC00);
+            }
 
-        // Convert one character at a time to avoid stack overflow on iPad.
-        //
-        // fromCharCode will not handle all legal Unicode values (up to 21 bits).  See
-        // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCharCode
-        for (var i = 0; i < length; ++i) {
-            result += String.fromCharCode(view[i]);
         }
         return result;
     };
 
+    function inRange(a, min, max) {
+        return min <= a && a <= max;
+    }
+
+    function utf8Handler(byte) {
+        // If bytes_needed = 0, then we are starting a new character
+        if (bytes_needed === 0) {
+            // 1 Byte Ascii character
+            if (inRange(byte, 0x00, 0x7F)) {
+                // Return a code point whose value is byte.
+                return byte;
+            }
+
+            // 2 Byte character
+            else if (inRange(byte, 0xC2, 0xDF)) {
+                bytes_needed = 1;
+
+                // Set UTF-8 code point to byte & 0x1F.
+                code_point = byte & 0x1F;
+            }
+
+            // 3 Byte character
+            else if (inRange(byte, 0xE0, 0xEF)) {
+                // If byte is 0xE0, set utf-8 lower boundary to 0xA0.
+                if (byte === 0xE0) {
+                    lower_boundary = 0xA0;
+                }
+                // If byte is 0xED, set utf-8 upper boundary to 0x9F.
+                if (byte === 0xED) {
+                    upper_boundary = 0x9F;
+                }
+
+                bytes_needed = 2;
+
+                // Set UTF-8 code point to byte & 0xF.
+                code_point = byte & 0xF;
+            }
+
+            // 4 Byte character
+            else if (inRange(byte, 0xF0, 0xF4)) {
+                // If byte is 0xF0, set utf-8 lower boundary to 0x90.
+                if (byte === 0xF0) {
+                    lower_boundary = 0x90;
+                }
+                // If byte is 0xF4, set utf-8 upper boundary to 0x8F.
+                if (byte === 0xF4) {
+                    upper_boundary = 0x8F;
+                }
+
+                bytes_needed = 3;
+
+                // Set UTF-8 code point to byte & 0x7.
+                code_point = byte & 0x7;
+            } else {
+                throw new RuntimeError('String decoding failed,');
+            }
+
+            return null;
```

Comment:
There's no reason to `return null;` here, you can just `return;`  We almost never use `null` in our code.